src/ml-os-em.jl

```diff
@@ -0,0 +1,140 @@
+# mlem.jl
+# ML-EM algortihm for SPECT reconstruction
+export mlem, mlem!, osem, osem!
+export Ablock
+using LinearMapsAA: LinearMapAO, LinearMapAA
+
+
+"""
+    mlem!(out, x0, ynoisy, background, A; niter = 20)
+Inplace version of ML-EM algorithm for SPECT reconstruction.
+- `out`: Output
+- `x0`: Initial guess
+- `ynoisy`: (Noisy) measurements
+- `background`: Background effects, e.g., scatters
+- `A`: System matrix
+- `niter`: Number of iterations
+"""
+function mlem!(out::AbstractArray,
+			   x0::AbstractArray,
+               ynoisy::AbstractArray,
+               background::AbstractArray,
+               A::Union{AbstractArray, LinearMapAO};
+               niter::Int = 20)
+    all(>(0), background) || throw("need background > 0")
+	size(out) != size(x0) && throw(DimensionMismatch("size out and x0 not match"))
+    asum = A' * ones(eltype(ynoisy), size(ynoisy)) # this allocates
+    asum[(asum .== 0)] .= Inf
+    ybar = similar(ynoisy)
+    yratio = similar(ynoisy)
+    back = similar(x0)
+	copyto!(out, x0)
+    time0 = time()
+    for iter = 1:niter
+        @show iter, extrema(out), time() - time0
+        mul!(ybar, A, out)
+        @. yratio = ynoisy / (ybar + background) # coalesce broadcast!
+        mul!(back, A', yratio) # back = A' * (ynoisy / ybar)
+        @. out *= back / asum # multiplicative update
+    end
+    return out
+end
+
+
+"""
+    mlem(x0, ynoisy, background, A; niter = 20)
+ML-EM algorithm for SPECT reconstruction.
+- `x0`: Initial guess
+- `ynoisy`: (Noisy) measurements
+- `background`: Background effects, e.g., scatters
+- `A`: System matrix
+- `niter`: Number of iterations
+"""
+mlem(x0::AbstractArray,
+	 ynoisy::AbstractArray,
+	 background::AbstractArray,
+	 A::Union{AbstractArray, LinearMapAO};
+	 kwargs...) = mlem!(similar(x0), x0, ynoisy, background, A; kwargs...)
+
+
+"""
+    Ablock(plan, nblocks)
+Generate a vector of linear maps for OSEM.
+-`plan`: SPECTrecon plan
+-`nblocks`: Number of blocks in OSEM
+"""
+function Ablock(plan::SPECTplan, nblocks::Int)
+    nx, ny, nz = size(plan.mumap)
+    nview = plan.nview
+    @assert rem(nview, nblocks) == 0 || throw("nview must be divisible by nblocks!")
```

Comment:
```suggestion
    rem(nview, nblocks) == 0 || throw("nview must be divisible by nblocks!")
```
we don't use both assert and throw together